doc/src/manual/noteworthy-differences.md

```diff
@@ -224,6 +229,24 @@ For users coming to Julia from R, these are some noteworthy differences:
   * The commonly used `Int` type corresponds to the machine integer type (`Int32` or `Int64`).
     This means it will overflow, such that `2^64 == 0`. If you need larger values use another appropriate type,
     such as `Int128`, [`BigInt`](@ref) or a floating point type like `Float64`.
+  * The imaginary unit `sqrt(-1)` is represented in Julia as `im`, not `j` as in Python.
+  * In Julia, the exponentiation operator is `^`, not `**` as in Python.
+  * In Julia uses `nothing` of type `Nothing` to represent null value, whereas Python uses `None` of type `NoneType`.
+  * In Julia, the standard operators over a matrix type are matrix operations, whereas, in Python, the standard operators are element-wise operations. When both `A` and `B` are matrices, `A * B` in Julia performs matrix multiplication, not element-wise multiplication as in Python. `A * B` in Julia is equivalent with `A @ B` in Python, whereas `A * B` in Python is equivalent with `A .* B` in Julia.
+  * The adjoint operator `'` in Julia returns an adjoint of a vector (a lazy representation of row vector), whereas the transpose operator `.T` over a vector in Python returns the original vector (non-op).
+  * Functions in Python have one implementation (no polymorphism), whereas in Julia a function may contain multiple concrete implementations (called *Methods*), selected via multiple dispatch.
+  * There are no classes in Julia. Instead they are structures (mutable or immutable), containing data but no methods. 
+  * Calling a method of a class in Python (`a = MyClass(x), x.func(y)`) corresponds to a function call in Julia, e.g. `a = MyStruct(x), func(x::MyStruct, y)`. In general, multiple dispatch is more flexible and powerful than the Python class system.
+  * Python classes can inherit from one or more superclasses. Julia structures may have exactly one abstract supertype. 
+  * The Python code structure is defined by directories (Packages) and files (Modules), whereas the logical Julia program structure (Packages and Modules) is independent of the file strucutre (`include` for additional files).
+  * Conditional assignment in Python `x = 1 if y>0 else -1` corresponds to the Ternary operator `x = y>0 ? 1 : -1` in Julia.
+  * Macros in Julia can do what decorators in Python do, and much more.
+  * Exception handling in Julia is done using `try` - `catch` - `finally`, instead of `try` - `except` - `finally`. In contrast to Python, it is not recommended to use exception handling as part of the normal workflow in Julia due to performance reasons.
+  * In Julia loops are fast, there is no need to write "vectorized" code for performance reasons.
+  * Non-constant global variables impose a significant performance loss in Julia and should therefore be avoided (see [Performance Tips](@ref man-performance-tips)). For Python, global variables are not associated with a performance penalty.  
```

Comment:
Technically, globals are slow in Python, too...

```python
In [1]: def sum1(xs):
   ...:     acc = 0
   ...:     for x in xs:
   ...:         acc += x
   ...:     return acc

In [2]: def sum2(xs):
   ...:     global acc
   ...:     acc = 0
   ...:     for x in xs:
   ...:         acc += x
   ...:     return acc

In [3]: xs = range(1000)

In [4]: %timeit sum1(xs)
39.8 µs ± 399 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [5]: %timeit sum2(xs)
59.3 µs ± 778 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

Reply:
So this claim is wrong since there is a performance penalty for globals in Python, it's just that locals are almost as slow as globals, so there's not as much of a penalty. Not sure how to word that best.

Reply:
Interesting, I was not aware of this speed difference in Python. Do you know the reason (type inference is required in any case)?

Suggestion for the text:

> For Python, global variables are associated with a much smaller performance penalty.

Reply:
Same as in Julia, presumably: the Python interpreter knows that a local can't be changed out from under it, allowing it to do certain work once rather than every time it accesses the variable.

Reply:
> Suggestion for the text:
> 
> > For Python, global variables are associated with a much smaller performance penalty.

I guess it's more like you don't care the performance penalty of globals in Python because the loops are slow in the first place?  Maybe the main message here is something like this?

> Be careful with non-constant global variables in Julia, especially in tight loops. Since you can write close-to-metal code in Julia (unlike Python), the effect of globals can be drastic (see Performance Tips).

Reply:
@tkf I like your suggestion, changing it accordingly.